doc/reference/kernel/other/async_notify.rst

```diff
@@ -0,0 +1,25 @@
+.. _async_notification:
+
```

Comment:
I propose to move this out of kernel/other and put it directly under doc/reference under a new category, for example doc/reference/services or whatever.